### ISSUE TITLE
list exercise error

### ISSUE THREAD

**Issue by Ibrahim-Islam** [scala-exercises/exercises-scalatutorial#34]
According to https://github.com/scala-exercises/exercises-scalatutorial/blob/master/src/main/scala/scalatutorial/sections/StandardLibrary.scala#L72-L97

```
val nums = Nil.::(4).::(3).::(2).::(1)

nums match {
  // Lists of `Int` that starts with `1` and then `2`
  case 1 :: 2 :: xs => …
  // Lists of length 1
  case x :: Nil => …
  // Same as `x :: Nil`
  case List(x) => …
  // The empty list, same as `Nil`
  case List() =>
  // A list that contains as only element another list that starts with `2`
  case List(2 :: xs) => …
}
```

the last case of `case List(2 :: xs)` gives error because the inferred type is `List[Int]` but the case signature is expecting `List[List[Int]]`. Please correct me if I am wrong. If I am not then how should it be re-written?

**Comment by julienrf** [comment]
Hi, it’s true that the example doesn’t compile. The idea was more to show various kinds of patterns.

**Comment by Ibrahim-Islam** [comment]
For someone new to Scala like me this could be a bit confusing. I think this can be refactored to show two lists or remove the last case altogether.

**Comment by FRosner** [comment]
I agree with @Ibrahim-Islam that any code should be copy-paste ready to try out and play with it. Maybe you want to add some useful return values, @Ibrahim-Islam?

**Comment by Ibrahim-Islam** [comment]
Yes and while at it maybe do a PR to make this right for future learners :)
But would like your opinion on how it should be corrected.

**Comment by FRosner** [comment]
@Ibrahim-Islam make a proposal :) Ping me if you like to discuss it :)
